flink-state-backends/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/RocksDBStateBackendConfigTest.java

```diff
@@ -172,6 +173,40 @@ public void testConfigureTimerService() throws Exception {
 		keyedBackend.dispose();
 	}
 
+	/**
+	 * Validates that user custom configuration from code should override the flink-conf.yaml.
+	 * @throws Exception
+	 */
+	@Test
+	public void testConfigureTimerServiceLoadingFromApplication() throws Exception {
+		final Environment env = getMockEnvironment(tempFolder.newFolder());
+
+		RocksDBStateBackend backend = new RocksDBStateBackend(tempFolder.newFolder().toURI().toString());
+		Configuration config = new Configuration();
+		config.setString(
+			RocksDBOptions.TIMER_SERVICE_FACTORY,
+			RocksDBStateBackend.PriorityQueueStateType.ROCKSDB.toString());
+		backend = backend.configure(config, Thread.currentThread().getContextClassLoader());
+
+		RocksDBKeyedStateBackend<Integer> keyedBackend = createKeyedStateBackend(backend, env);
+
+		Assert.assertEquals(RocksDBPriorityQueueSetFactory.class, keyedBackend.getPriorityQueueFactory().getClass());
```

Comment:
In these two lines we duplicate check from `testConfigureTimerService`.
I would suggest we test only one thing in a test at a time.